src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,37 @@
+package duke.command;
+
+import duke.Storage;
+import duke.TaskList;
+import duke.Ui;
+import duke.task.Task;
+
+/**
+ * Command that adds new tasks into the TaskList.
+ */
+public class AddCommand extends Command {
+
+    /** Task to be added. */
+    private Task task;
+
+    /**
+     * Creates a command to add task into TaskList.
+     * @param task Task to be added.
+     */
+    public AddCommand(Task task) {
+        this.task = task;
```

Comment:
Perhaps `this` is unnecessary? The basic coding standard doesn't require this, but is suggested by the advanced style guide.
```suggestion
        task = task;
```